pkg/cloud/vsphere/vsphere_cloud_config/config.go

```diff
@@ -0,0 +1,44 @@
+package vsphere_cloud_config
+
+import (
+	"fmt"
+
+	"k8s.io/klog/v2"
+
+	"gopkg.in/yaml.v2"
+)
+
+// ReadConfig parses vSphere cloud-config file and returns CPIConfig structure
+// Accepts both YAML and INI formats as input.
+// YAML format takes precedence, in case parsing YAML is not successful function falls back to the legacy INI format.
+// Unlike 'cloud-provider-vsphere' version of a similar function, this does ignore environment variables.
+func ReadConfig(config []byte) (*CPIConfig, error) {
+	if len(config) == 0 {
+		return nil, fmt.Errorf("vSphere config is empty")
+	}
+
+	klog.V(3).Info("Try to parse vSphere config, yaml format first")
+	cfg, err := readCPIConfigYAML(config)
+	if err != nil {
+		klog.Warningf("Parsing yaml config failed, fallback to ini: %s", err)
```

Comment:
I think the idiom for errors is to use `%v` in logs and `%w` when in a `fmt.Errorf`
```suggestion
		klog.Warningf("Parsing yaml config failed, fallback to ini: %v", err)
```